flocker/node/agents/cinder.py

```diff
@@ -305,8 +340,8 @@ def detach_volume(self, blockdevice_id):
 
         # This'll blow up if the volume is deleted from elsewhere.  FLOC-1882.
         wait_for_volume(
-            volume_manager=self.nova_volume_manager,
-            expected_volume=nova_volume,
+            volume_manager=self.cinder_volume_manager,
```

Comment:
Switching to `cinder_api` instead of instantiating `CinderBlockAPI` revealed that `wait_for_volume` was calling `list` which isn't declared in the `INovaVolumeManager` interface.
So I now use the cinder client to get the volume and to wait for it's status to change.